### PR TITLE
[BOOKINGSG-5979][WX]add new component SelectHistogram

### PR DESCRIPTION
**Pull request by wx-uf**
1. add new component SelectHistogram in DS
2. add storybook and test cases accordingly

### CODE REVIEW

**Review by choo95dev — APPROVED** [review]

File: src/select-histogram/types.ts [hunk 1]
```diff
@@ -0,0 +1,16 @@
+import { HistogramSliderProps } from "../histogram-slider";
+import { TruncateType } from "../shared/dropdown-list-v2/types";
+import { DropdownAlignmentType } from "../shared/dropdown-wrapper";
+
+export interface SelectHistogramSliderProps {
```
Comment (qroll) [inline src/select-histogram/types.ts:5]: to also include the following optional props:
- `className`
- `readOnly`
- `onBlur` - can be typed without the event arg as `() => void`

File: src/select-histogram/types.ts [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+    error?: boolean | undefined;
+    alignment?: DropdownAlignmentType | undefined;
+    dropdownZIndex?: number | undefined;
+    histogramSlider: HistogramSliderProps;
```
Comment (qroll) [inline src/select-histogram/types.ts:9]: can pick specific props instead as not all are used or relevant

i.e.
`Pick<HistogramSliderProps, "bins" | "interval" | "renderEmptyView"> `

and move `onChange`, `rangeLabelPrefix` and `rangeLabelSuffix` to `SelectHistogramSliderProps`

File: src/select-histogram/select-histogram.tsx
Comment (qroll) [inline src/select-histogram/select-histogram.tsx:1]: to also include the `focused` handling with `onBlur` callback

File: src/select-histogram/select-histogram.tsx [hunk 3]
```diff
@@ -0,0 +1,201 @@
...
+            {getDisplayValue()}
+        </LabelContainer>
+    );
+
+    const renderElement = () => {
+        return (
+            <InputBox
+                className={className}
+                data-testid={"select-histogram-selector"}
+                id="select-histogram-selector"
```
Comment (qroll) [inline src/select-histogram/select-histogram.tsx:146]: don't hardcode, use the props instead

**Review by qroll — CHANGES_REQUESTED** [review]

**Review by qroll — APPROVED** [review]

**Pull request merged**